src/main/java/org/folio/service/routinglists/RoutingListService.java

```diff
@@ -1,51 +1,120 @@
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.PoLine;
 import org.folio.rest.jaxrs.model.RoutingList;
+import org.folio.rest.jaxrs.model.RoutingListCollection;
+import org.folio.service.UserService;
+import org.folio.service.orders.PurchaseOrderLineService;
+import org.folio.service.routinglists.validators.RoutingListValidatorUtil;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static org.folio.orders.utils.ResourcePathResolver.ORDER_SETTINGS;
+import static org.folio.orders.utils.ResourcePathResolver.ROUTING_LISTS;
+import static org.folio.orders.utils.ResourcePathResolver.TEMPLATE_REQUEST;
+import static org.folio.orders.utils.ResourcePathResolver.resourcesPath;
 
 public class RoutingListService {
+  private static final Logger logger = LogManager.getLogger(RoutingListService.class);
 
-  private static final Logger log = LogManager.getLogger();
-  private static final UUID TEMPLATE_REQUEST_ID = UUID.fromString("9465105a-e8a1-470c-9817-142d33bc4fcd");
-  private static final String TEMPLATE_REQUEST_LANG = "en";
-  private static final String TEMPLATE_REQUEST_OUTPUT = "text/html";
   private static final String ROUTING_LIST_ENDPOINT = resourcesPath(ROUTING_LISTS);
   private static final String ORDER_SETTINGS_ENDPOINT = resourcesPath(ORDER_SETTINGS);
   private static final String ROUTING_USER_ADDRESS_TYPE_ID = "ROUTING_USER_ADDRESS_TYPE_ID";
   private static final String ROUTING_LIST_BY_ID_ENDPOINT = ROUTING_LIST_ENDPOINT + "/{id}";
+  private static final String ROUTING_LIST_BY_POL_ID = "poLineId==%s";
+
   private static final String TEMPLATE_REQUEST_ENDPOINT = resourcesPath(TEMPLATE_REQUEST);
+  private static final UUID TEMPLATE_REQUEST_ID = UUID.fromString("9465105a-e8a1-470c-9817-142d33bc4fcd");
+  private static final String TEMPLATE_REQUEST_LANG = "en";
+  private static final String TEMPLATE_REQUEST_OUTPUT = "text/html";
 
-  private final RestClient restClient;
+  private final PurchaseOrderLineService poLineService;
   private final UserService userService;
+  private final RestClient restClient;
 
-  public RoutingListService(RestClient restClient, UserService userService) {
+  public RoutingListService(RestClient restClient, PurchaseOrderLineService poLineService, UserService userService) {
     this.restClient = restClient;
+    this.poLineService = poLineService;
     this.userService = userService;
   }
 
+  public Future<RoutingList> getRoutingList(String rListId, RequestContext requestContext) {
+    RequestEntry requestEntry = new RequestEntry(ROUTING_LIST_BY_ID_ENDPOINT).withId(rListId);
+    return restClient.get(requestEntry, RoutingList.class, requestContext);
+  }
+
+  public Future<Void> updateRoutingList(RoutingList routingList, RequestContext requestContext) {
+    try {
+      validateRoutingList(routingList, requestContext);
+    }
+    catch (HttpException e) {
```

Comment:
per code convention catch should be on the same level as closing bracket

Reply:
Got it